sys/net/link_layer/ieee802154/submac.c

```diff
@@ -539,35 +540,49 @@ int ieee802154_set_phy_conf(ieee802154_submac_t *submac, uint16_t channel_num,
 int ieee802154_set_rx(ieee802154_submac_t *submac)
 {
     ieee802154_fsm_state_t current_state = submac->fsm_state;
-
-    if (current_state == IEEE802154_FSM_STATE_RX) {
-        return -EALREADY;
-    }
-    else if (current_state != IEEE802154_FSM_STATE_IDLE
-             || ieee802154_submac_process_ev(submac,
-                                             IEEE802154_FSM_EV_REQUEST_SET_RX_ON) !=
-             IEEE802154_FSM_STATE_RX) {
-        return -EBUSY;
+    ieee802154_fsm_state_t next_state;
+    int res = -EBUSY;
+
+    switch (current_state) {
+        case IEEE802154_FSM_STATE_RX:
```

Comment:
```suggestion
    switch (current_state) {
    case IEEE802154_FSM_STATE_RX:
```

`uncrustify` got this right :wink: